clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```
@@ -0,0 +1,485 @@
+	metricVal  [][]float64
+	metricLock sync.Mutex
+
+	podPairNumber int
+
+	podPairCh chan uniquePodPair
+
+	// startTimeStampForTestExecution is a futureTimeStamp sent to all pods to start
+	// sending the traffic at the same time.
+	startTimeStampForTestExecution int64
+
+	resultSummary testResultSummary
+}
+
+func (npm *networkPerformanceMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		klog.Error("error starting action: ", err)
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		if err = npm.validate(config); err != nil {
+			klog.Error("error in validation action: ", err)
+			return nil, err
+		}
+		if err = npm.start(config); err != nil {
+			klog.Error("error starting measurement: ", err)
+			return nil, err
+		}
+	case "gather":
+		summary, err := npm.gather()
+		if err != nil && !errors.IsMetricViolationError(err) {
+			klog.Error("error gathering metrics: ", err)
+			return nil, err
+		}
+		klog.Info("metric: ", summary)
+		return []measurement.Summary{summary}, err
+	default:
+	}
+	return nil, nil
+}
+
+func (npm *networkPerformanceMeasurement) start(config *measurement.Config) error {
+	npm.initialize(config)
+
+	npm.namespace = netperfNamespace
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return err
+	}
+
+	if err := npm.createClusterRoleBinding(); err != nil {
+		return err
+	}
+
+	if err := npm.createCRD(); err != nil {
+		return err
+	}
+
+	if err := npm.createWorkerPods(); err != nil {
+		return err
+	}
+
+	if err := npm.waitForWorkerPodsReady(); err != nil {
+		return err
+	}
+
+	npm.storeWorkerPods()
+
+	npm.executeTest()
+
+	go npm.watchNetworkTestEvents()
+
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) executeTest() {
+	switch npm.podRatioType {
+	case oneToOne, manyToMany:
+		npm.execNToMTest()
+	default:
+		klog.Error("Invalid Pod Ratio")
```

this should return an error - not log it

---

Yes made the changes.